portal-web/test/functional/com/liferay/portalweb/macros/Theme.macro

```diff
@@ -69,6 +69,16 @@ definition {
 		}
 	}
 
+	macro deployThemeWar {
+		var liferayHome = PropsUtil.get("liferay.home.dir.name");
+		var testBaseDirName = PropsUtil.get("test.base.dir.name");
+		var testDependenciesDirName = PropsUtil.get("test.dependencies.dir.name");
+
+		ArchiveUtil.archive("${testBaseDirName}/${testDependenciesDirName}/${themeWar}", "${liferayHome}/deploy/${themeWar}");
+
+		Pause(locator1 = "10000");
```

Comment:
Are we able to avoid an implicit pause here? Maybe we can have the test wait for console log text for the theme to load? I was looking at this with Calvin in the App Manager macro but I think we kept running into issues trying to get that assertion to work for a while. Have you tried something like that? https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/macros/AppManager.macro#L66-L74

Reply:
We don't have a function called IsConsoleTextPresent, so actually the while condition here https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/macros/AppManager.macro#L66-L74 doesn't check the console text, it just do pausing 3s 15 times. 

I tried to add a new function as we have the selenium method isConsoleTextPresent, but I didn't make it work. If I used like this:`while (!(IsConsoleTextPresent(value1 = "STARTED ${themeName}")))`
An error about missing locator1 property will thrown, but if I changed to use lcoator1, an NPE error thrown when running test. I assume that maybe the selenium method need to be updated in order to use it.

Reply:
I think we can read the console text from log file directly, I know there are tests that also doing this. I will update the macro and rerun ci test.

Reply:
ok sounds good, maybe can ask in poshi slack channel if anyone has ideas or tips on this.

Reply:
@kenjiheigel are we using the `IsConsoleTextPresent` function correctly? It sounds like it's broken or can be deprecated if we're using something else to assert console log text.

Reply:
Updated macro in commit https://github.com/john-co/liferay-portal/pull/315/commits/c959d5732790624afd33b326a243d367bc2ed651, rutest in PR https://github.com/zhangyan0701/liferay-portal/pull/216.